Project Infra (#12)
squid-cache/squid-cache.github.io: docs/ProjectInfrastructure/DailySnapshotsGeneration.md

# How daily snapshots files are generated

As of the 2023-11-02, this is the process to generate snapshot tarballs.

1. Jenkins jobs run at regular cadences, monitoring the repository, run a full build test.
   [`trunk-arm64-matrix`](https://build.squid-cache.org/job/trunk-arm64-matrix/) is used for the
   VCS ["master" branch](https://github.com/squid-cache/squid/tree/master/),
   while a version specific `N-matrix` (e.g. `6-matrix` or `5-matrix`) are used for the stable and old releases
1. if these jobs are successful, they will trigger the corresponding tarball creation job
   (e.g. [`website-tarballs-head`](https://build.squid-cache.org/job/website-tarballs-head/) for the `trunk_amd64_matrix`).

>    The tarball jobs trust that if they are invoked, it's because the branch has a stable build and testing result.

Reviewer: For many, "stable result" means "the same result over several tests (including the same failure)". I am guessing that we want to describe a different condition:

```suggestion
   The tarball creation jobs trust that they are only invoked for the branch that has passed build tests.
```

or, more accurately and to the point:

```suggestion
   The tarball creation jobs do not verify whether the branch passes build tests.
```
